src/Sprites/spriteMap.ts

```diff
@@ -0,0 +1,627 @@
+        let t = RawTexture.CreateRGBATexture(
+        floatArray,
+        this.options.stageSize.x,
+        this.options.stageSize.y,
+        this._scene,
+        false,
+        false,
+        Texture.NEAREST_NEAREST,
+        Engine.TEXTURETYPE_FLOAT
+        );
+
+        return t;
+    }
+
+    /**
+    * Modfies the data of the tileMaps
+    * @param _layer is the ID of the layer you want to edit on the SpriteMap
+    * @param pos is the iVector2 Coordinates of the Tile
+    * @param tile The SpriteIndex of the new Tile
+    */
+    changeTiles(_layer: number= 0, pos: any , tile: number= 0): void {
+
+        let buffer: any = [];
+        buffer = this._tileMaps[_layer]!._texture!._bufferView;
+        if (!buffer) {
+            return;
+        }
+
+        let p = new Array();
+        if (pos instanceof Vector2) {
+            p.push(pos);
+        }else {
+            p = pos;
+        }
+
+        for (let i = 0; i < p.length; i++) {
+            let _p = p[i];
+            _p.x = Math.floor(_p.x);
+            _p.y = Math.floor(_p.y);
+            let id = (_p.x * 4) + (_p.y * (this.options.stageSize.x * 4));
+            buffer[id] = tile;
+        }
+
+        let t = this.createTileBuffer(buffer);
+        this._tileMaps[_layer].dispose();
+        this._tileMaps[_layer] = t;
+        this._material.setTextureArray("tileMap", this._tileMaps);
+    }
+
+    /**
+    * Creates the animationMap texture Buffer
+    * @param buffer normaly and array of numbers, or a false to generate from scratch
+    * @returns RawTexture of the animationMap
+    */
+    createTileAnimationBuffer(buffer: any): RawTexture {
+        let data = new Array();
+        if (!buffer) {
+            for (let i = 0; i < this.spriteCount; i++) {
+                data.push(0, 0, 0, 0);
+                let count = 1;
+                while (count < (this.options.maxAnimationFrames || 4)) {
+                    data.push(0, 0, 0, 0);
+                    count++;
+                }
+            }
+        }else {
+            data = buffer;
+        }
+
+        let floatArray = new Float32Array(data);
+        let t = RawTexture.CreateRGBATexture(
+        floatArray,
+        this.spriteCount,
+        (this.options.maxAnimationFrames || 4),
+        this._scene,
+        false,
+        false,
+        Texture.NEAREST_NEAREST,
+        Engine.TEXTURETYPE_FLOAT
+        );
+
+        return t;
+    }
+    /**
+    * Modfies the data of the animationMap
+    * @param cellID is the Index of the Sprite
+    * @param _frame is the target Animation frame
+    * @param toCell is the Target Index of the next frame of the animation
+    * @param time is a value between 0-1 that is the trigger for when the frame should change tiles
+    * @param speed is a global scalar of the time varable on the map.
+    */
+    addAnimationToTile(cellID: number= 0, _frame: number= 0, toCell: number= 0, time: number= 0, speed: number = 1): void {

```

Comment:
All the utilities function should be flag as private and not exposed publicly (keep the API simple)